File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/frontend/src/grid-in-a-template.js
/*
 * Copyright (C) 2024 Vaadin Ltd
 *
 * This program is available under Vaadin Commercial License and Service Terms.
 *
 * See {@literal <https://vaadin.com/commercial-license-and-service-terms>}  for the full
 * license.
 */
import { PolymerElement } from '@polymer/polymer/polymer-element.js';
import { html } from '@polymer/polymer/lib/utils/html-tag.js';

class GridInATemplate extends PolymerElement {
  static get template() {
    return html`
        <vaadin-grid id="grid"></vaadin-grid>
`;
  }

  static get is() {
      return 'grid-in-a-template'
  }
}
customElements.define(GridInATemplate.is, GridInATemplate);

File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/grid/it/GridInATemplate.java
/*
 * Copyright (C) 2024 Vaadin Ltd
 *
 * This program is available under Vaadin Commercial License and Service Terms.
 *
 * See {@literal <https://vaadin.com/commercial-license-and-service-terms>}  for the full
 * license.
 */
package com.vaadin.flow.component.grid.it;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.HtmlImport;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.grid.Grid;
import com.vaadin.flow.component.polymertemplate.Id;
import com.vaadin.flow.component.polymertemplate.PolymerTemplate;
import com.vaadin.flow.templatemodel.TemplateModel;

@Tag("grid-in-a-template")
@HtmlImport("src/grid-in-a-template.html")
@JsModule("src/grid-in-a-template.js")
public class GridInATemplate extends PolymerTemplate<TemplateModel> {

    @Id
    Grid<String> grid;

    public Grid<String> getGrid() {
        return grid;
    }

}
